### ISSUE TITLE
using enum struct as parameter when it's declaring

### ISSUE THREAD

**Issue by Starbish** [alliedmodders/sourcepawn#307]
`
enum struct PlayerTable
{
	int a[4];

	void Test(PlayerTable test)
	{
//		test.a[3] = 2;
	}
}`

Compiler says "error 159: brackets after variable name indicate a fixed-size array, but size could not be determined - either specify sizes, an array initializer, or use dynamic syntax (such as 'char[] x')"

But i doubt it's an invalid expression. Is this a bug? or It's just an invalid expression?

What i tried to make was to check the data between two different PlayerTable to confirm it's identical or not.

like this,
`
enum struct PlayerTable
{
	PlayerTableIdentifier pTableIdentifier;
	char tableAuthIdFieldName[MAX_FIELD_NAME_LENGTH];
	AuthIdType tableAuthIdType;
	int tableDataState[MAXPLAYERS+1];
	ArrayList loadWaitingList;

	bool IsIdenticalTo(PlayerTable pTableBuffer){

		if(!StrEqual(pTableBuffer.dbPresetName, this.pTableIdentifier.dbPresetName))
			return false;
		if(!StrEqual(pTableBuffer.tableName, this.pTableIdentifier.tableName))
			return false;
		if(!StrEqual(pTableBuffer.tableAuthIdFieldName, this.tableAuthIdFieldName))
			return false;
		if(pTableBuffer.tableAuthIdType != this.tableAuthIdType)
			return false;

		return true;
	}
}
`

**Comment by Starbish** [comment]
```
enum struct PlayerTable2
{
    int a[4];
    void Test(PlayerTable2[1] test)
    {
        PrintToServer("%i", test[0].a[3]);
    }
}

public void OnPluginStart()
{
    PlayerTable2 x[1];
    x[0].a[3] = 123;
    x[0].Test(x);
}
```

One of SM discord user found that this expression worked as expected. Will I be able to use this expression even though it's not recommended?

**Comment by dvander** [comment]
Yes unfortunately that is the issue. I would like to fix this but I haven't nailed down how yet.

In the meantime, it is better to have a global method, like "ComparePlayers(Player a, Player b)".
